### DISCUSSION TITLE
Azure AD SSO for users and pipelines

### DISCUSSION THREAD

**Discussion by Petteroe** [vercel/community#5767] [Feature Question] [answered]
### Question

Hi,

Noob here. I apologize in advance if my search skills were not up to par and this has been answered before.

We are a large company with our main infra setup in Azure and GitHub. This means all people registered in Azure AD, SSO with GitHub, use of Azure KeyVault for governing secrets etc. 

We are very keen on using Vercel, but our security team is concerned about the administration and governance of identities and secrets:

- Single sign-on from Azure AD, including automatic removal when we remove them from our Azure AD
- No personal access tokens in the GitHub integrations
- Secrets managed in the same (or similar) fashion as today (Azure KeyVault)
- Managed identity or similar for authenticating against our backend services in Azure
- Our Vercel apps behind our common Akamai gateway so that we can get these apps into the same monitoring and response regime as our other stuff

These are the biggest concerns that I want to be able to answer so that we can get going with Vercel. Have these issues been raised before, and do you have solutions for them or approaches that have been accepted by similarly demanding customers?

Thanks!

**Comment by amyegan** [comment, marked as answer]
Hi @Petteroe! I'm more familiar with Hobby and Pro than I am with Enterprise exclusive features, and I don't want to give you the wrong info. The best thing would be to [reach out to our sales team](https://vercel.com/contact/sales) who can set up a call to discuss your questions in depth. 

I'll let the team know to be on the lookout for a message from you. In the meantime, these pages have more info about our security features and Enterprise plans:
- [Vercel Enterprise](https://vercel.com/enterprise)
- [Security at Vercel](https://vercel.com/security)
- [SAML Single Sign-On](https://vercel.com/docs/security/saml#)
- [Sensitive Environment Variables](https://vercel.com/docs/projects/environment-variables/sensitive-environment-variables#)
- [Access Roles](https://vercel.com/docs/accounts/team-members-and-roles/access-roles#)

**Reply by Petteroe** [reply]
Thank you for pointing me in the right direction @amyegan. I will read through the links you list here so that I can ask the sales team the correct questions :)
